ios/browser/api/history/brave_history_api.mm

```diff
@@ -0,0 +1,309 @@
+#include "components/sync/driver/sync_service.h"
+
+#include "ios/chrome/browser/application_context.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state_manager.h"
+#include "ios/chrome/browser/browsing_data/browsing_data_remove_mask.h"
+#include "ios/chrome/browser/browsing_data/browsing_data_remover.h"
+#include "ios/chrome/browser/browsing_data/browsing_data_remover_factory.h"
+#include "ios/chrome/browser/history/history_service_factory.h"
+#include "ios/chrome/browser/history/web_history_service_factory.h"
+#include "ios/chrome/browser/sync/profile_sync_service_factory.h"
+#include "ios/web/public/thread/web_task_traits.h"
+#include "ios/web/public/thread/web_thread.h"
+
+#include "net/base/mac/url_conversions.h"
+#include "ui/base/page_transition_types.h"
+#include "url/gurl.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+#pragma mark - IOSHistoryNode
+
+@interface IOSHistoryNode () {
+  std::u16string title_;
+  GURL gurl_;
+  base::Time date_added_;
+}
+@end
+
+@implementation IOSHistoryNode
+
+- (instancetype)initWithURL:(NSURL*)url
+                      title:(NSString* _Nullable)title
+                  dateAdded:(NSDate* _Nullable)dateAdded {
+  if ((self = [super init])) {
+    [self setUrl:url];
+
+    if (title) {
+      [self setTitle:title];
+    }
+
+    if (dateAdded) {
+      [self setDateAdded:dateAdded];
+    }
+  }
+
+  return self;
+}
+
+- (void)setUrl:(NSURL*)url {
+  gurl_ = net::GURLWithNSURL(url);
+}
+
+- (NSURL*)url {
+  return net::NSURLWithGURL(gurl_);
+}
+
+- (void)setTitle:(NSString*)title {
+  title_ = base::SysNSStringToUTF16(title);
+}
+
+- (NSString*)title {
+  return base::SysUTF16ToNSString(title_);
+}
+
+- (void)setDateAdded:(NSDate*)dateAdded {
+  date_added_ = base::Time::FromDoubleT([dateAdded timeIntervalSince1970]);
+}
+
+- (NSDate*)dateAdded {
+  return [NSDate dateWithTimeIntervalSince1970:date_added_.ToDoubleT()];
+}
+@end
+
+#pragma mark - BraveHistoryAPI
+
+@interface BraveHistoryAPI () <BraveHistoryDriverDelegate> {
+  // Browser State Used to retreive History Driver - History Service and Web
+  // History Service
+  ChromeBrowserState* browser_state_;
+  // History Browsing Service Driver
+  std::unique_ptr<BraveBrowsingHistoryDriverIOS> browsing_history_driver_;
+  // History Browsing Service
+  std::unique_ptr<history::BrowsingHistoryService> browsing_history_service_;
+  // History Service for adding and querying
+  history::HistoryService* history_service_;
+  // WebhistoryService for remove and elete operations
+  history::WebHistoryService* web_history_service_;
+  // Tracker for history requests.
+  base::CancelableTaskTracker tracker_;
+}
+@property(nonatomic, strong) void (^query_completion)(NSArray<IOSHistoryNode*>*)
+    ;
+@end
+
+@implementation BraveHistoryAPI
+
+/// Shared Singleton Instance for History API
```

Comment:
We should not be copying this from the bookmarks api because it's not safe to have a singleton like this.  See https://github.com/brave/brave-core/commit/a5c5a1f1de8c55b7f4119b40f23efae1d569dd79#diff-c59f38b8a49e62dafe5064d748612f3a1a32215373ef69968b2cbe7f7cc94949R115

Reply:
Singleton Instances are removed and API lasses are added as property in brave_core_main. Constructors are changed to require browser state while initializing. 

Fix at https://github.com/brave/brave-core/pull/8869/commits/5d5d66f281595d929b896fddc1bf20d3fd1c92ac